WordPress/src/main/java/org/xmlrpc/android/ApiHelper.java

```diff
@@ -1190,4 +1014,54 @@ else if (attrName.equals("href"))
         }
         return null; // never found the rsd tag
     }
+
+    /*
+     * fetches a single post saves it to the db - note that this should NOT be called from main thread
+     */
+    public static boolean updateSinglePost(int localBlogId, String remotePostId, boolean isPage) {
+        Blog blog = WordPress.getBlog(localBlogId);
+        if (blog == null || TextUtils.isEmpty(remotePostId)) {
+            return false;
+        }
+
+        XMLRPCClientInterface client = XMLRPCFactory.instantiate(
+                blog.getUri(),
+                blog.getHttpuser(),
+                blog.getHttppassword());
+
+        Object[] apiParams;
+        if (isPage) {
+            apiParams = new Object[]{
+                    blog.getRemoteBlogId(),
+                    remotePostId,
+                    blog.getUsername(),
+                    blog.getPassword()
+            };
+        } else {
+            apiParams = new Object[]{
+                    remotePostId,
+                    blog.getUsername(),
+                    blog.getPassword()
+            };
+        }
+
+        try {
+            Object result = client.call(isPage ? "wp.getPage" : "metaWeblog.getPost", apiParams);
+
+            if (result != null && result instanceof Map) {
+                Map postMap = (HashMap) result;
+                List<Map<?, ?>> postsList = new ArrayList<>();
+                postsList.add(postMap);
+
+                WordPress.wpDB.savePosts(postsList, localBlogId, isPage, true);
```

Comment:
I believe I found the problem. I was requesting the newly published post from the server before the existing post was updated in the local db (with the new remote id), resulting in dups due to the different IDs. And after looking at it, I decided we should request the uploaded post even if it wasn't a local draft to make sure the local copy matches the server.

Both of these are resolved in dd046e5.